sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
+      return create();
+    }
+    final String[] args = properties.stringPropertyNames()
+      .stream()
+      .map(k -> String.format("--%s=%s", k, properties.getProperty(k)))
+      .toArray(String[]::new);
+    return fromArgs(args).create();
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties
+   * filtered with the specified prefix.
+   */
+  public static PipelineOptions fromProperties(final String prefix, final Properties properties) {
+    if (prefix == null) {
+      return fromProperties(properties);
+    }
+    if (properties == null) {
+      return create();
+    }
+    final Properties instantiationProperties = properties.stringPropertyNames()
+      .stream()
+      .filter(k -> k.startsWith(prefix))
+      .collect(
+        Properties::new,
+        (p, k) -> p.setProperty(k.substring(prefix.length()), properties.getProperty(k)),
+        Hashtable::putAll);
+    return fromProperties(instantiationProperties);
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @return a pipeline options instance based on current environment and overriden with
+   * system properties key/values and filtered with a custom prefix.
+   */
+  public static PipelineOptions fromJvm(final String prefix) {
+    final Properties properties = System.getenv()
+      .entrySet()
+      .stream()
+      .collect(
+              Properties::new,
+              (p, e) -> p.setProperty(e.getKey(), e.getKey()),
+              Properties::putAll);
+    properties.putAll(System.getProperties());
+    return fromProperties(prefix, properties);
+  }
+
+  /**
+   * @return a pipeline options instance based on current environemnt and system properties
+   * filtered based on <code>beam.</code> prefix.
+   */
+  public static PipelineOptions fromJvm() {
```

Comment:
Don't add this. Each use needs to choose its own prefix. The test pipeline can read from `beam.TestPipelineOptions` and that prefix should be owned over there, not here. Any user should always have to specify a prefix on the system properties.

Reply:
I don't think so, it is very important to propose a default prefix (beam.) to enable frameworks to have a default and uniform implicit usage. This doesn't mean we should encapsulate the prefix but that we should have an imlpicit prefix when the integration framework don't care much. If not done each framework will have its own prefix which wouldn't help much users in several cases.

wdyt?

Reply:
PipelineOptions are an explicit per-job property. Only a program that runs exactly one job could use this, and that program can just specify a prefix. Again, this is different to the discussion on dev. If you want to set up global default pipeline options via properties that is a separate thing that just supporting parsing them. Bring back to the dev list.

Reply:
Not really, test flavor proves it is not about only one job. Also it concerns only the submit so even with the one job constraint it concers a lot of users so is worth it IMHO. Will add it in the mail, thanks.